batch.go

```diff
@@ -15,6 +15,7 @@ type (
 	BatchIterator interface {
 		Done() bool
 		Next(messageID string, opts *BatchOptions) (*EventBatch, error)
+		NextWithPartition(opts *BatchOptions) (map[string]*EventBatch, error)
```

Comment:
What would you think about `Next()` batching by partition by default?

A batch by definition must only contain events targeting a single partition. If someone calls `Next` (in its current state, unaware of batching) not knowing the rules of batching, then they will meet with an error at some point. By making `Next` batch aware by default, then it will release the consumer of the responsibility from understanding batching rules and move the responsibility to the SDK. I think this will push consumers of the batch iterator down a path of success.

Reply:
> A batch by definition must only contain events targeting a single partition.

With the above rule, I agree I should improve the `Next()` method instead of creating a new one.

Reply:
One more concern here. If I just change the `Next(eventID string, opts *BatchOptions)` method, I also need to change its parameters since `eventID` is not enough for a map of batches. So I should treat this as a break change for users, right?

Reply:
In `Next`, you should return a single batch which is filled with events targeting a single partition. `Next` should return EOF upon running out of batches.

```go
func Next(...) {
  // do I have any partitions of events in my map[string][]*event
  if hasKeys(map) {
    // build batch from first partition []*events
    return partitionBatch
  }
  
  return EOF
}
```

What do you think?